packages/dts-test/defineComponent.test-d.tsx

```diff
@@ -293,17 +293,17 @@ describe('with object props', () => {
     />,
   )
 
-  expectType<Component>(
-    <MyComponent
-      b="b"
-      dd={{ n: 1 }}
-      ddd={['ddd']}
-      eee={() => ({ a: 'eee' })}
-      fff={(a, b) => ({ a: a > +b })}
-      hhh={false}
-      jjj={() => ''}
-    />,
-  )
+  // expectType<Component>(
+  //   <MyComponent
+  //     b="b"
+  //     dd={{ n: 1 }}
+  //     ddd={['ddd']}
+  //     eee={() => ({ a: 'eee' })}
+  //     fff={(a, b) => ({ a: a > +b })}
+  //     hhh={false}
+  //     jjj={() => ''}
+  //   />,
+  // )
```

Comment:
This does not make much sense to me, should the JSX.Element be considered a Vue component? 

It was working previously because the type of `Component` was pretty much `any` because of LegacyComponentOptions https://github.com/vuejs/core/blob/main/packages/runtime-core/src/componentOptions.ts#L471